module/zfs/dsl_deadlist.c

```diff
@@ -1028,6 +1028,17 @@ dsl_process_sub_livelist(bpobj_t *bpobj, bplist_t *to_free, zthr_t *t,
 	    .t = t
 	};
 	int err = bpobj_iterate_nofree(bpobj, dsl_livelist_iterate, &arg, size);
+	if (err == EINTR) {
```

Comment:
I think that we only expect `bpobj_iterate_nofree()` to return either 0 or EINTR, and in the case of 0, the avl should be empty.  So this code is correct.

That said, it seems like if it were possible to get a different error from `bpobj_iterate_nofree()`, it would make sense for this layer to return it up to the caller, rather than panicking.  We could do that by doing something like:
```c
int err = bpobj_iterate_nofree(bpobj, dsl_livelist_iterate, &arg, size);
VERIFY(err != 0 || avl_numnodes(&avl) == 0);
... avl_destroy_nodes, unconditionally
avl_destroy(&avl);
return (err);
```

Reply:
Applied